Documentation/rfc/v3api.proto

```diff
@@ -0,0 +1,276 @@
+  // The key expiration generates an event in event history.
+  rpc LeaseCreate(LeaseCreateRequest) returns (LeaseCreateResponse) {}
+
+  // LeaseRevoke revokes a lease. All the key attached to the lease will be expired and deleted.
+  rpc LeaseRevoke(LeaseRevokeRequest) returns (LeaseRevokeResponse) {}
+
+  // LeaseAttach attaches keys with a lease.
+  rpc LeaseAttach(LeaseAttachRequest) returns (LeaseAttachResponse) {}
+
+  // LeaseTnx likes Tnx. It does one additional thing that all the keys put into the
+  // store are attached with the given lease. This is rpc is useful when you want to
+  // put a key and attach it to a lease atomically.
+  rpc LeaseTnx(LeaseTnxRequest) returns (LeaseTnxResponse) {}
+
+  // KeepAlive keeps the lease alive.
+  rpc LeaseKeepAlive(stream LeaseKeepAliveRequest) returns (stream LeaseKeepAliveResponse) {}
+}
+
+message RequestHeader {
+  // the cluster id to check with
+  // if the cluster id is not matched, an error is returned.
+  optional uint64 cluster_id = 1;
+}
+
+message ResponseHeader {
+  // an error type message?
+  optional string error = 1;
+  optional uint64 cluster_id = 2;
+  optional uint64 member_id = 3;
+  // index of the store when the request was applied.
+  optional int64 index = 4;
+  // term of raft when the request was applied.
+  optional uint64 raft_term = 5;
+}
+
+message RangeRequest {
+  // if the range_end is not given, the request returns the key.
+  optional bytes key = 1;
+  // if the range_end is given, it gets the keys in range [key, range_end).
+  optional bytes range_end = 2;
+  // limit the number of keys returned.
+  optional int64 limit = 3;
+  // the response will be consistent with previous request with same token if the token is 
+  // given and is vaild.
+  optional int64 consistent_token = 4;
+}
+
+message RangeResponse {
+  optional ResponseHeader header = 1;
+  repeated KeyValue kvs = 2;
+  optional int64 consistent_token = 3;
+}
+
+message PutRequest {
+  optional bytes key = 1;
+  optional bytes value = 2;
+}
+
+message PutResponse {
+  optional ResponseHeader header = 1;
+}
+
+message DeleteRangeRequest {
+  // if the range_end is not given, the request deletes the key.
+  optional bytes key = 1;
+  // if the range_end is given, it deletes the keys in range [key, range_end).
+  optional bytes range_end = 2;
+}
+
+message DeleteRangeResponse {
+  optional ResponseHeader header = 1;
+}
+
+message RequestUnion {
+  oneof value {
+    RangeRequest request_range = 1;
+    PutRequest request_put = 2;
+    DeleteRequest request_delete = 3;
+  }
+}
+
+message ResponseUnion {
+  oneof value {
+    RangeResponse request_range = 1;
+    PutResponse request_put = 2;
+    DeleteResponse request_delete = 3;
+  }
+}
+
+message Compare {
+  enum CompareType {
+    equal = 0;
+    greater = 1;
+    less = 2;
+  }
+  optional CompareType type = 1;
+  // key path
+  optional bytes key = 2;
+  one of {
```

Comment:
`oneof` I think -- along with a type. More generally, this file should actually work under protoc